axlearn/common/attention.py

```diff
@@ -1912,23 +1918,46 @@ def _forward_for_mode(
         return dict(i_proj=i_proj_state), output
 
     def _logit_biases_for_mask(
-        self, *, mode: ForwardMode, seq_len: int, time_step: Optional[Tensor] = None
+        self,
+        *,
+        mode: ForwardMode,
+        kv_len: int,
+        query_len: Optional[int] = None,
+        time_step: Optional[Tensor] = None,
     ) -> Optional[Tensor]:
         """Returns the configured attention mask in the form of logit biases.
 
         ... or None if the implementation of _compute_attention supports applying masks natively.
 
-        For (ForwardMode.FORWARD, ForwardMode.INIT_STATES), the mask can be broadcast to
-        [batch, num_heads, seq_len, seq_len].
+        Args:
+            mode: The forward propagation mode, chosen from
+                (ForwardMode.FORWARD, ForwardMode.INIT_STATES, ForwardMode.EXTEND_STEP).
+            kv_len: The sequence length. For (ForwardMode.INIT_STATES, ForwardMode.EXTEND_STEP),
+                this is equal to the KV cache size.
+            query_len: Only used for (ForwardMode.FORWARD, ForwardMode.INIT_STATES).
+                If set, this is the query length. Otherwise, it uses kv_len as the query length.
```

Comment:
```suggestion
                If set, this is the query length. Otherwise, it uses kv_len as the query length.
                Must be None for ForwardMode.EXTEND_STEP.
```